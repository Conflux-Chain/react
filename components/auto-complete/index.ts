import AutoComplete, { useAutoCompleteHandle } from './auto-complete'
export { useAutoCompleteHandle }
export default AutoComplete
