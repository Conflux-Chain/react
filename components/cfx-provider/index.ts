import CfxProvider from './cfx-provider'

export default CfxProvider
