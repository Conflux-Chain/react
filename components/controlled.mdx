If you think the data flow in react carefully.

uncontrolled component:
caller do not know the internal state of the component

controlled component:
the state of called component is fully controlled by `props` passed in 

The above two are identified by react offical team. However, there is also something 
in between, which can not fit in the patterns mentioned above

third patten: the called need to know the callee state sometims but not always
think form as an example, the caller only need to know the state of callee when 'submit'
happen rather than keep track of the component all the time

forth pattern: the caller need to set the callee state forcelly on a one time, not inter
