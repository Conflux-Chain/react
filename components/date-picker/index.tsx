import { Moment } from 'moment'
import momentGenerateConfig from '@jnoodle/rc-picker/lib/generate/moment'
import generatePicker, {
  PickerProps,
  PickerDateProps,
  RangePickerProps as BaseRangePickerProps,
} from './generate-picker'

export type DatePickerProps = PickerProps<Moment>
export type MonthPickerProps = Omit<PickerDateProps<Moment>, 'picker'>
export type WeekPickerProps = Omit<PickerDateProps<Moment>, 'picker'>
export type RangePickerProps = BaseRangePickerProps<Moment>

export const DatePicker = generatePicker<Moment>(momentGenerateConfig)

export default DatePicker
