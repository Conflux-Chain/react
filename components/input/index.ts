import Input, { useInputHandle } from './input'

export { useInputHandle }
export default Input
