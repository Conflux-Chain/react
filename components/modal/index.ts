import Modal from './modal'
import ModalTitle from './modal-title'
import ModalSubtitle from './modal-subtitle'
import ModalContent from './modal-content'
import ModalAction from './modal-action'
import useModalHandle from './use-imperative-modal'

Modal.Title = ModalTitle
Modal.Subtitle = ModalSubtitle
Modal.Content = ModalContent
Modal.Action = ModalAction
Modal.useModalHandle = useModalHandle

export default Modal
