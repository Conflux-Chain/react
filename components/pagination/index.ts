import Pagination from './pagination'
import PaginationPrevious from './pagination-previous'
import PaginationNext from './pagination-next'
import usePaginationHandle from './use-pagination-handle'
Pagination.Previous = PaginationPrevious
Pagination.Next = PaginationNext
Pagination.usePaginationHandle = usePaginationHandle
export default Pagination
