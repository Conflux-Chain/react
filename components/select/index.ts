import Select from './select'
import SelectOption from './select-option'
import useSelectHandle from './use-imperactive-select'

Select.Option = SelectOption
Select.useSelectHandle = useSelectHandle

export default Select
