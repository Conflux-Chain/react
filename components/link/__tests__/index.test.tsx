import React from 'react'
import { mount } from 'enzyme'
import { ExternalLink } from '@zeit-ui/react-icons'
import { Link } from 'components'

describe('Link', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <div>
        <Link href="https://react.zeit-ui.co">link</Link>
        <Link href="https://react.zeit-ui.co" underline>
          link
        </Link>
        <Link href="https://react.zeit-ui.co" block>
          link
        </Link>
        <Link href="https://react.zeit-ui.co" disabled>
          link
        </Link>
        <Link href="https://react.zeit-ui.co" plain>
          link
        </Link>
        <Link href="https://react.zeit-ui.co" iconLeft={<ExternalLink />}>
          link
        </Link>
        <Link href="https://react.zeit-ui.co" iconRight={<ExternalLink />}>
          link
        </Link>
        <Link
          href="https://react.zeit-ui.co"
          iconLeft={<ExternalLink />}
          iconRight={<ExternalLink />}>
          link
        </Link>
      </div>,
    )
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should be no errors when href missing', () => {
    const errorSpy = jest.spyOn(console, 'error')
    const wrapper = mount(<Link />)
    expect(errorSpy).not.toHaveBeenCalled()
    expect(() => wrapper.unmount()).not.toThrow()
    errorSpy.mockRestore()
  })

  it('should forward ref', () => {
    let ref = React.createRef<HTMLAnchorElement>()
    const errorSpy = jest.spyOn(console, 'error')
    const wrapper = mount(<Link ref={ref} />)

    expect(errorSpy).not.toHaveBeenCalled()
    expect(ref.current).not.toBeNull()
    expect(() => wrapper.unmount()).not.toThrow()
    errorSpy.mockRestore()
  })

  it('an warning should be thrown when using the pure prop', () => {
    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {})
    const link = mount(<Link pure />)
    expect(errorSpy).toHaveBeenCalled()
    expect(() => link.unmount()).not.toThrow()
    errorSpy.mockRestore()
  })

  it('an info should be shown when using the color prop', () => {
    const errorSpy = jest.spyOn(console, 'info').mockImplementation(() => {})
    const link = mount(<Link color />)
    expect(errorSpy).toHaveBeenCalled()
    expect(() => link.unmount()).not.toThrow()
    errorSpy.mockRestore()
  })

  it('an info should be shown when using the icon prop', () => {
    const errorSpy = jest.spyOn(console, 'info').mockImplementation(() => {})
    const link = mount(<Link icon />)
    expect(errorSpy).toHaveBeenCalled()
    expect(() => link.unmount()).not.toThrow()
    errorSpy.mockRestore()
  })
})
