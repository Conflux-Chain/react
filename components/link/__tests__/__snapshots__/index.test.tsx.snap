// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `
"<div><a href=\\"https://react.zeit-ui.co\\" class=\\"link \\">link<style>
          .link {
            display: inline-flex;
            align-items: baseline;
            padding: 0;
            width: fit-content;
            line-height: inherit;
            text-decoration: none;
            color: #0054fe;
            border-radius: 0;
            transition: color 200ms ease 0ms;
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: none;
          }

          .link:hover {
            color: #0433dc;
            background-color: unset;
          }

          .link.disabled {
            color: #9b9eac;
            pointer-events: none;
            cursor: default;
          }

          .link :global(.left) {
            margin-left: 0;
          }

          .link :global(.right) {
            margin-right: 0;
          }
        </style></a><a href=\\"https://react.zeit-ui.co\\" class=\\"link \\">link<style>
          .link {
            display: inline-flex;
            align-items: baseline;
            padding: 0;
            width: fit-content;
            line-height: inherit;
            text-decoration: none;
            color: #0054fe;
            border-radius: 0;
            transition: color 200ms ease 0ms;
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: underline;
          }

          .link:hover {
            color: #0433dc;
            background-color: unset;
          }

          .link.disabled {
            color: #9b9eac;
            pointer-events: none;
            cursor: default;
          }

          .link :global(.left) {
            margin-left: 0;
          }

          .link :global(.right) {
            margin-right: 0;
          }
        </style></a><a href=\\"https://react.zeit-ui.co\\" class=\\"link \\">link<style>
          .link {
            display: inline-flex;
            align-items: baseline;
            padding: 2px 4px;
            width: fit-content;
            line-height: inherit;
            text-decoration: none;
            color: #0054fe;
            border-radius: 4px;
            transition: color 200ms ease 0ms;
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: none;
          }

          .link:hover {
            color: #0433dc;
            background-color: #e8f4ff;
          }

          .link.disabled {
            color: #9b9eac;
            pointer-events: none;
            cursor: default;
          }

          .link :global(.left) {
            margin-left: 0;
          }

          .link :global(.right) {
            margin-right: 0;
          }
        </style></a><a href=\\"https://react.zeit-ui.co\\" class=\\"link  disabled\\">link<style>
          .link {
            display: inline-flex;
            align-items: baseline;
            padding: 0;
            width: fit-content;
            line-height: inherit;
            text-decoration: none;
            color: #0054fe;
            border-radius: 0;
            transition: color 200ms ease 0ms;
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: none;
          }

          .link:hover {
            color: #0433dc;
            background-color: unset;
          }

          .link.disabled {
            color: #9b9eac;
            pointer-events: none;
            cursor: default;
          }

          .link :global(.left) {
            margin-left: 0;
          }

          .link :global(.right) {
            margin-right: 0;
          }
        </style></a><a href=\\"https://react.zeit-ui.co\\" class=\\"link \\">link<style>
          .link {
            display: inline-flex;
            align-items: baseline;
            padding: 0;
            width: fit-content;
            line-height: inherit;
            text-decoration: none;
            color: inherit;
            border-radius: 0;
            transition: color 200ms ease 0ms;
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: none;
          }

          .link:hover {
            color: #0433dc;
            background-color: unset;
          }

          .link.disabled {
            color: #9b9eac;
            pointer-events: none;
            cursor: default;
          }

          .link :global(.left) {
            margin-left: 0;
          }

          .link :global(.right) {
            margin-right: 0;
          }
        </style></a><a href=\\"https://react.zeit-ui.co\\" class=\\"link \\"><span class=\\"icon left\\"><svg viewBox=\\"0 0 24 24\\" width=\\"24\\" height=\\"24\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" style=\\"color: currentColor;\\"><path d=\\"M18 13v6a2 2 0 01-2 2H5a2 2 0 01-2-2V8a2 2 0 012-2h6\\"></path><path d=\\"M15 3h6v6\\"></path><path d=\\"M10 14L21 3\\"></path></svg><style>
        .icon {
          margin: 0 5px;
          display: inline-flex;
          align-self: center;
          color: currentColor;
          width: 1rem;
        }
      </style></span>link<style>
          .link {
            display: inline-flex;
            align-items: baseline;
            padding: 0;
            width: fit-content;
            line-height: inherit;
            text-decoration: none;
            color: #0054fe;
            border-radius: 0;
            transition: color 200ms ease 0ms;
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: none;
          }

          .link:hover {
            color: #0433dc;
            background-color: unset;
          }

          .link.disabled {
            color: #9b9eac;
            pointer-events: none;
            cursor: default;
          }

          .link :global(.left) {
            margin-left: 0;
          }

          .link :global(.right) {
            margin-right: 0;
          }
        </style></a><a href=\\"https://react.zeit-ui.co\\" class=\\"link \\">link<span class=\\"icon right\\"><svg viewBox=\\"0 0 24 24\\" width=\\"24\\" height=\\"24\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" style=\\"color: currentColor;\\"><path d=\\"M18 13v6a2 2 0 01-2 2H5a2 2 0 01-2-2V8a2 2 0 012-2h6\\"></path><path d=\\"M15 3h6v6\\"></path><path d=\\"M10 14L21 3\\"></path></svg><style>
        .icon {
          margin: 0 5px;
          display: inline-flex;
          align-self: center;
          color: currentColor;
          width: 1rem;
        }
      </style></span><style>
          .link {
            display: inline-flex;
            align-items: baseline;
            padding: 0;
            width: fit-content;
            line-height: inherit;
            text-decoration: none;
            color: #0054fe;
            border-radius: 0;
            transition: color 200ms ease 0ms;
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: none;
          }

          .link:hover {
            color: #0433dc;
            background-color: unset;
          }

          .link.disabled {
            color: #9b9eac;
            pointer-events: none;
            cursor: default;
          }

          .link :global(.left) {
            margin-left: 0;
          }

          .link :global(.right) {
            margin-right: 0;
          }
        </style></a><a href=\\"https://react.zeit-ui.co\\" class=\\"link \\"><span class=\\"icon left\\"><svg viewBox=\\"0 0 24 24\\" width=\\"24\\" height=\\"24\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" style=\\"color: currentColor;\\"><path d=\\"M18 13v6a2 2 0 01-2 2H5a2 2 0 01-2-2V8a2 2 0 012-2h6\\"></path><path d=\\"M15 3h6v6\\"></path><path d=\\"M10 14L21 3\\"></path></svg><style>
        .icon {
          margin: 0 5px;
          display: inline-flex;
          align-self: center;
          color: currentColor;
          width: 1rem;
        }
      </style></span>link<span class=\\"icon right\\"><svg viewBox=\\"0 0 24 24\\" width=\\"24\\" height=\\"24\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" style=\\"color: currentColor;\\"><path d=\\"M18 13v6a2 2 0 01-2 2H5a2 2 0 01-2-2V8a2 2 0 012-2h6\\"></path><path d=\\"M15 3h6v6\\"></path><path d=\\"M10 14L21 3\\"></path></svg><style>
        .icon {
          margin: 0 5px;
          display: inline-flex;
          align-self: center;
          color: currentColor;
          width: 1rem;
        }
      </style></span><style>
          .link {
            display: inline-flex;
            align-items: baseline;
            padding: 0;
            width: fit-content;
            line-height: inherit;
            text-decoration: none;
            color: #0054fe;
            border-radius: 0;
            transition: color 200ms ease 0ms;
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: none;
          }

          .link:hover {
            color: #0433dc;
            background-color: unset;
          }

          .link.disabled {
            color: #9b9eac;
            pointer-events: none;
            cursor: default;
          }

          .link :global(.left) {
            margin-left: 0;
          }

          .link :global(.right) {
            margin-right: 0;
          }
        </style></a></div>"
`;
